### PR TITLE
Improve slider

### PR DESCRIPTION
**Pull request by Looooong**
+ Fixed the bug where dragging mouse outside `document.body` causes the slider to stop moving.
+ Changed event handler to capture pointer instead of registering/unregistering on `document.body` to handle slider dragging.
+ Improved styling by using design system utilitity classes instead of arbitrary styles. This depends on #13

### CODE REVIEW

**Comment by CreepGin** [comment]
Cool thx! Yes Pointer Events is the better way. I included the old implementation (renamed) back in for reference purposes. 

And `inset: false,` needs to be removed from `onejs-tw-config.js` for the `right-0` and `bottom-0` to work, right? I'll update that in a separate commit.

**Comment by Looooong** [comment]
> And `inset: false,` needs to be removed from `onejs-tw-config.js` for the `right-0` and `bottom-0` to work, right? I'll update that in a separate commit.

Ah yes, I forgot about the `inset: false` stuff. I overrided `inset: true` in my own project.

File: onejs/comps/Slider.tsx [hunk 1]
```diff
@@ -1,10 +1,71 @@
...
+            e.currentTarget.ReleasePointer(e.pointerId)
+        }
+        onPointerUp?.(e)
+    }, [onPointerUp])
+
+    const handlerPointerEvent = useCallback((e: IPointerEvent) => {
+        const width = trackRef.current.ve.layout.width
+        const ratio = math.saturate(e.localPosition.x / width)
+        activeTrackRef.current.style.width = `${Math.round(ratio * 100)}%`
+        onChange?.(math.lerp(min ?? 0, max ?? 0, ratio))
```
Comment (Looooong) [inline onejs/comps/Slider.tsx]: @CreepGin Looks like we need to use `max ?? 1` instead. However, to reduce the probablity of coding error in the future, we should include this code in the component body, before the `useEffect`:
```ts
  min ??= 0
  max ??= 1
  value ??= min
```
And then we don't have to bother with null-coalescing for the rest of the code. It still works correctly with hooks and dependency list.

**Comment by Looooong** [comment]
One more thing, in order to use pointer capturing, `ScriptEngine` must includes the `UnityEngine.UIElements.PointerCaptureHelper` extensions class.

File: onejs/comps/Slider.tsx (hunk 1 above)
Comment (CreepGin) [reply onejs/comps/Slider.tsx]: Oh the closure will mess up the type checking:
image
So I'll adjust it a little bit:
```tsx
const min = _min ?? 0
const max = _max ?? 1
const value = _value ?? min
```

**Comment by CreepGin** [comment]
> One more thing, in order to use pointer capturing, `ScriptEngine` must includes the `UnityEngine.UIElements.PointerCaptureHelper` extensions class.

Yup, should be included by default already.

Gonna merge this one now to sync up with `main`'s jsx.d.ts and onejs-tw-config.js

File: onejs/comps/Slider.tsx (hunk 1 above)
Comment (Looooong) [reply onejs/comps/Slider.tsx]: That's weird. I don't get that kind of error on my end.

**Comment by CreepGin** [comment]
Oh do you have "strictNullChecks" off by any chance? Though it should be on
by the default tsconfig.json

File: onejs/comps/Slider.tsx (hunk 1 above)
Comment (CreepGin) [reply onejs/comps/Slider.tsx]: Oh do you have "strictNullChecks" off by any chance?

Comment (Looooong) [reply onejs/comps/Slider.tsx]: Ah, I have `strictNullChecks` off.

**Pull request merged**